tests/unit/developers-dao.test.js

```diff
@@ -0,0 +1,174 @@
+    it('getDeveloperById should be fulfilled with a single result', () => {
+      const fulfilledCases = [
+        { expectedResult: {} },
+      ];
+
+      const fulfilledPromises = [];
+      _.each(fulfilledCases, ({ expectedResult }) => {
+        createConnStub({ rows: [expectedResult] });
+        const result = developersDao.getDeveloperById(fakeId);
+        fulfilledPromises.push(result.should
+          .eventually.be.fulfilled
+          .and.deep.equal(expectedResult));
+      });
+      return Promise.all(fulfilledPromises);
+    });
+
+    it('getDeveloperById should be rejected when multiple values are returned', () => {
+      const rejectedCases = [
+        { testCase: { rows: [{}, {}] }, error: 'Expect a single object but got multiple results' },
+      ];
+
+      const rejectedPromises = [];
+      _.each(rejectedCases, ({ testCase, error }) => {
+        createConnStub(testCase);
+
+        const result = developersDao.getDeveloperById('fakeId');
+        rejectedPromises.push(result.should
+          .eventually.be.rejectedWith(Error, error));
+
+        sinon.restore();
+      });
+      return Promise.all(rejectedPromises);
+    });
+  });
+
+  describe('Test postDeveloper', () => {
+    it('postDeveloper with improper body should be rejected', () => {
+      createConnStub();
+
+      const result = developersDao.postDeveloper('fakeId', 'fakeBody');
+      return result.should
+        .eventually.be.rejected
+        .and.be.an.instanceOf(TypeError);
+    });
+
+    it('postDeveloper should be fulfilled with singleResult', () => {
+      const fakeBody = {
+        data: {
+          attributes: {
+            name: 'test',
+          },
+        },
+      };
+
+      const expectedResult = {};
+      createConnStub({ rows: [{}], outBinds: { outId: 1 } });
+      const result = developersDao.postDeveloper(fakeBody);
+      return result.should
+        .eventually.be.fulfilled
+        .and.deep.equal(expectedResult);
+    });
+
+    const testCases = [
+      {
+        testCase: [],
+        expectedError: 'Cannot read property \'outId\' of undefined',
+        testDescription: 'outId is not returned',
+      },
+      {
+        testCase: { outBinds: { outId: 'fakeId' } },
+        expectedError: 'Cannot read property \'length\' of undefined',
+        testDescription: 'outId is returned without an additional response',
+      },
+    ];
+    _.forEach(testCases, ({ testCase, expectedError, testDescription }) => {
+      it(`postDeveloper should be rejected when ${testDescription}`, () => {
+        createConnStub(testCase);
+
+        const fakeBody = { data: { attributes: 'fakeAttributes' } };
+        const result = developersDao.postDeveloper(fakeBody);
+        return result.should
+          .eventually.be.rejectedWith(Error, expectedError);
+      });
+    });
+  });
+
+  describe('Test deleteDeveloper', () => {
+    it('deleteDeveloper should be fulfilled with single result', () => {
```

Comment:
The 404 for delete and patch are handled in `{developerId}.js`. Currently `deleteDeveloper` just returns the value returned from `conn.execute()`. Should I refactor so that 404s are handled in `deleteDeveloper`?